### DISCUSSION TITLE
Converting mlua::Value type to rust types and variable number of inputs

### DISCUSSION THREAD

**Discussion by AlexXi19** [mlua-rs/mlua#251] [General]
I'm trying to override the build in lua print function to better print tables, how can i best do this? I'm having some trouble converting mlua::Value to a rust type, I looked into serializing but no success. I was also wondering if it's possible to create a function that takes a variable number of inputs, I've tried making the input a vector and that didn't work. Thanks in advance for the help!

**Comment by khvzak** [comment]
>  I'm having some trouble converting mlua::Value to a rust type,

Could you show your code?

>  I was also wondering if it's possible to create a function that takes a variable number of inputs,

You probably need [`MultiValue`](https://docs.rs/mlua/0.8.7/mlua/struct.MultiValue.html) or [`Variadic`](https://docs.rs/mlua/0.8.7/mlua/struct.Variadic.html) for this

**Reply by AlexXi19** [reply]
So far I've tried the `serialize` function and using debug printing with `println!("{:?}, arg)` but both print this: `String(String(Ref(7)))`. Similar outputs for other types of values.

**Reply by khvzak** [reply]
An example with serializing:
```rust
    let print = lua.create_function(|lua, mut args: mlua::MultiValue| {
        while let Some(val) = args.pop_front() {
            let val = lua.from_value::<serde_json::Value>(val)?;
            println!("{}", serde_json::to_string_pretty(&val).unwrap());
        }
        Ok(())
    })?;
    lua.globals().set("print", print)?;
    lua.load(r#"
        print({a = 1, b = 2}, {"hello", "world"})
    "#).exec().unwrap();
```
